src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs

```diff
@@ -48,11 +48,14 @@ public bool IsCompatible(NuGetFramework target, NuGetFramework candidate)
             // check the cache for a solution
             var cacheKey = new CompatibilityCacheKey(target, candidate);
 
-            bool? result = _cache.GetOrAdd(cacheKey, (Func<CompatibilityCacheKey, bool>)((key)
-                =>
-            { return IsCompatibleCore(target, candidate) == true; }));
+            bool result;
+            if (!_cache.TryGetValue(cacheKey, out result))
```

Comment:
The following should work too: 

```suggestion
            if (!_cache.TryGetValue(cacheKey, out bool result))
```

Reply:
actually, don't accept the suggestion as it'll cause a compile error, but you get the idea.